feat: implement generic CIM extraction framework

amarnath-ac opened the pull request:
- Add comprehensive CIMExtractorFramework implementation for unified property extraction
- Add Description property extraction from CIM_ComputerSystem.Description
- Add HostName property extraction from CIM_ComputerSystem.DNSHostName
- Add Status property extraction from CIM_Processor
- Update test entities and validation for complete property coverage

Output:
 curl -k -u "standalone:G@ppm0ym" http://localhost:8181/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163 | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   438  100   438    0     0     89      0  0:00:04  0:00:04 --:--:--    93
{
  "@odata.context": "/redfish/v1/$metadata#ComputerSystem.ComputerSystem",
  "@odata.id": "/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163",
  "@odata.type": "#ComputerSystem.v1_26_0.ComputerSystem",
  "HostName": null,
  "Id": "a72605c3-7494-4c63-855b-273410b4e163",
  "Manufacturer": "Intel(R) Client Systems",
  "Model": "NUC12WSKv5",
  "Name": "a72605c3-7494-4c63-855b-273410b4e163",
  "PowerState": "On",
  "SerialNumber": "BTWS22800956",
  "SystemType": "Physical"
}
// Note: CIM_ComputerSystem properties (Description, HostName, Status) are not available via AMT WSMAN interface and will remain empty in the response.
CIM_ComputerSystemPackage and CIM_SystemPackaging are empty with "curl -s http://localhost:8181/api/v1/amt/hardwareInfo/a72605c3-7494-4c63-855b-273410b4e163 | jq .

amarnath-ac committed:
feat: implement generic CIM extraction framework

- Add comprehensive CIMExtractorFramework implementation for
  unified property extraction
- Add Description property extraction from CIM_ComputerSystem.Description
- Add HostName property extraction from CIM_ComputerSystem.DNSHostName
- Add Status property extraction from CIM_Processor
- Update test entities and validation for complete property coverage

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc commented on redfish/openapi/merged/redfish-openapi.yaml:
```diff
@@ -1539,13 +1549,6 @@ components:
             that cannot be conveyed with other properties defined for the Placement
             object.
           x-versionAdded: v1_7_0
-        FacilityName:
```
When i auto-generated these fields are coming back. If you need to remove them then they need be removed from Resource.v1_23_0.yaml file

amarnath-ac replied:
In my system auto generate is removing these changes, as discussed you will push the changes to restore these changes.

sudhir-intc commented on redfish/internal/usecase/computer_system.go:
```diff
@@ -65,40 +65,31 @@ func (uc *ComputerSystemUseCase) GetComputerSystem(ctx context.Context, systemID
 	// Create OData fields following the reference pattern
 	odataContext := generated.OdataV4Context("/redfish/v1/$metadata#ComputerSystem.ComputerSystem")
-	odataType := generated.OdataV4Type("#ComputerSystem.v1_22_0.ComputerSystem")
+	odataType := generated.OdataV4Type("#ComputerSystem.v1_26_0.ComputerSystem")
```
move this to a constant.

amarnath-ac replied:
done

sudhir-intc commented on redfish/internal/usecase/wsman_repo.go:
```diff
@@ -37,27 +74,390 @@ var (
+}
+
+// ExtractPropertyFromHardwareInfo extracts a single property from pre-fetched hardware info.
+func (f *CIMExtractorFramework) ExtractPropertyFromHardwareInfo(hwInfo dto.HardwareInfo, config CIMPropertyConfig) interface{} {
```
This function is not used in any other module other than here. can you make this a private function i.e rename ExtractPropertyFromHardwareInfo to extractPropertyFromHardwareInfo

amarnath-ac replied:
done

sudhir-intc commented on redfish/internal/usecase/wsman_repo.go:
```diff
@@ -37,27 +74,390 @@ var (
+}
+
+// ExtractProperty extracts a single property from CIM data based on configuration.
+func (f *CIMExtractorFramework) ExtractProperty(ctx context.Context, systemID string, config CIMPropertyConfig) interface{} {
```
This function is not used in any other module other than here. can you make this a private function i.e rename ExtractProperty to extractProperty

amarnath-ac replied:
done

sudhir-intc commented on redfish/internal/usecase/wsman_repo.go:
```diff
@@ -37,27 +74,390 @@ var (
+}
+
+// ExtractMultipleProperties extracts multiple properties in a single call for efficiency.
+func (f *CIMExtractorFramework) ExtractMultipleProperties(ctx context.Context, systemID string, configs []CIMPropertyConfig) map[string]interface{} {
```
This function is not used in any other module other than here. can you make this a private function i.e rename ExtractMultiplePropertiesto extractMultipleProperties

amarnath-ac replied:
done

sudhir-intc commented on redfish/internal/usecase/wsman_repo.go:
```diff
@@ -149,26 +599,11 @@ func (r *WsmanComputerSystemRepo) GetByID(ctx context.Context, systemID string)
-	// Extract hardware info for manufacturer, model, serial number
-	manufacturer, model, serialNumber := r.extractHardwareInfo(ctx, systemID)
+	// Extract CIM data using the global configuration with static transformers
+	cimData := r.extractCIMData(ctx, systemID, allCIMConfigs)
```
rename the function to getCIMProperties - as extractCIMData is misleading because it doesn't actually extract; it delegates to ExtractMultipleProperties which does the real extraction AND fetches hardware info

amarnath-ac replied:
done

sudhir-intc reviewed (dismissed):
incorporate the changes suggested

amarnath-ac committed:
refactor: extract hardcoded strings to constants and improve function visibility

- Extract schema references to constants in computer_system.go
- Convert exported CIM extraction methods to private functions
